Duplicate key -100 in GrpcServerProcessor after upgrading to Quarkus 3.20.0

After upgrading from Quarkus 3.17.8 to 3.20.0 (LTS), our application fails to build due to a Duplicate key error during gRPC server initialization. This issue does not occur in 3.17.8 and appears to be a regression or a stricter validation introduced in 3.20.0.

Stack Trace (excerpt):
```
Build step io.quarkus.grpc.deployment.GrpcServerProcessor#initializeServer threw an exception: 
java.lang.IllegalStateException: Duplicate key -100 (attempted merging values Runtime proxy of interface io.vertx.core.Handler with id proxykey89 and Runtime proxy of interface io.vertx.core.Handler with id proxykey280)
```

**Steps to Reproduce**
Use Quarkus 3.20.0 with quarkus-grpc and multiple gRPC services.
Build the project using mvn clean install.
Observe failure during the GrpcServerProcessor build step.

**Expected Behavior** 

The application should build successfully as it does in Quarkus 3.17.8.

**Actual Behavior** 

Build fails with IllegalStateException due to duplicate proxy key registration.

**Environment**

Quarkus version: 3.20.0
Java version: 21
gRPC: quarkus-grpc
Build tool: Maven

**Additional Notes**
No duplicate @GrpcService implementations for the same proto service.
No custom Vert.x Handler beans registered manually.
Issue appears to be related to CDI proxy generation or Vert.x handler registration.
Works fine in 3.17.8.

Hi Groaned,

Please find the entire trace:

[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------< com.hughes:websocket >------------------------
[INFO] Building websocket 2.2.0
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- quarkus:3.20.0:dev (default-cli) @ websocket ---
[INFO] Invoking jacoco:0.8.12:prepare-agent (default-prepare-agent) @ websocket
[INFO] argLine set to -javaagent:/Users/venkatasandeep/.m2/repository/org/jacoco/org.jacoco.agent/0.8.12/org.jacoco.agent-0.8.12-runtime.jar=destfile=/Users/venkatasandeep/Main/apeem/apeWS/target/jacoco-quarkus.exec,append=true,exclclassloader=*QuarkusClassLoader
[INFO] Invoking resources:3.3.1:resources (default-resources) @ websocket
[INFO] Copying 2 resources from src/main/resources to target/classes
[INFO] Invoking quarkus:3.20.0:generate-code (default) @ websocket
[INFO] [io.quarkus.grpc.codegen.GrpcCodeGen] Successfully finished generating and post-processing sources from proto files
[INFO] Invoking compiler:3.14.0:compile (default-compile) @ websocket
[INFO] Recompiling the module because of changed source code.
[INFO] Compiling 105 source files with javac [debug release 21] to target/classes
[INFO] Annotation processing is enabled because one or more processors were found
  on the class path. A future release of javac may disable annotation processing
  unless at least one processor is specified by name (-processor), or a search
  path is specified (--processor-path, --processor-module-path), or annotation
  processing is enabled explicitly (-proc:only, -proc:full).
  Use -Xlint:-options to suppress this message.
  Use -proc:none to disable annotation processing.
[INFO] Invoking resources:3.3.1:testResources (default-testResources) @ websocket
[INFO] skip non existing resourceDirectory /Users/venkatasandeep/Main/apeem/apeWS/src/test/resources
[INFO] Invoking quarkus:3.20.0:generate-code-tests (default) @ websocket
[INFO] Invoking compiler:3.14.0:testCompile (default-testCompile) @ websocket
[INFO] Recompiling the module because of changed dependency.
[INFO] Compiling 7 source files with javac [debug release 21] to target/test-classes
[INFO] Annotation processing is enabled because one or more processors were found
  on the class path. A future release of javac may disable annotation processing
  unless at least one processor is specified by name (-processor), or a search
  path is specified (--processor-path, --processor-module-path), or annotation
  processing is enabled explicitly (-proc:only, -proc:full).
  Use -Xlint:-options to suppress this message.
  Use -proc:none to disable annotation processing.
[WARNING] [io.quarkus.deployment.dev.DevModeCommandLineBuilder] Changed debug port to 64362 because of a port conflict
Listening for transport dt_socket at address: 64362
2025-07-08 08:01:37,588 WARN  [io.qua.hib.orm.dep.HibernateOrmProcessor] (build-13) Could not find a suitable persistence unit for model classes:
        - io.quarkus.hibernate.orm.panache.PanacheEntity
        - io.quarkus.hibernate.orm.panache.PanacheEntityBase
2025-07-08 08:01:38,317 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2025-07-08 08:01:38,496 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.grpc.deployment.GrpcServerProcessor#initializeServer threw an exception: java.lang.IllegalStateException: Duplicate key -100 (attempted merging values Runtime proxy of interface io.vertx.core.Handler with id proxykey104 and Runtime proxy of interface io.vertx.core.Handler with id proxykey284)
        at java.base/java.util.stream.Collectors.duplicateKeyException(Collectors.java:135)
        at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:182)
        at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at io.quarkus.grpc.deployment.GrpcServerProcessor.initializeServer(GrpcServerProcessor.java:721)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at java.base/java.lang.Thread.run(Thread.java:1583)
        at org.jboss.threads.JBossThread.run(JBossThread.java:499)

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:372)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:289)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:61)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:89)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:428)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:107)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:70)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.grpc.deployment.GrpcServerProcessor#initializeServer threw an exception: java.lang.IllegalStateException: Duplicate key -100 (attempted merging values Runtime proxy of interface io.vertx.core.Handler with id proxykey104 and Runtime proxy of interface io.vertx.core.Handler with id proxykey284)
        at java.base/java.util.stream.Collectors.duplicateKeyException(Collectors.java:135)
        at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:182)
        at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at io.quarkus.grpc.deployment.GrpcServerProcessor.initializeServer(GrpcServerProcessor.java:721)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at java.base/java.lang.Thread.run(Thread.java:1583)
        at org.jboss.threads.JBossThread.run(JBossThread.java:499)

        at io.quarkus.builder.Execution.run(Execution.java:122)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:78)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:161)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:368)
        ... 9 more
Caused by: java.lang.IllegalStateException: Duplicate key -100 (attempted merging values Runtime proxy of interface io.vertx.core.Handler with id proxykey104 and Runtime proxy of interface io.vertx.core.Handler with id proxykey284)
        at java.base/java.util.stream.Collectors.duplicateKeyException(Collectors.java:135)
        at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:182)
        at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
--
Tests paused
Press [e] to edit command line args (currently ''), [r] to resume testing, [o] Toggle test output, [:] for the terminal, [h] for more options>

--- a/extensions/websockets/server/deployment/src/main/java/io/quarkus/websockets/deployment/ServerWebSocketProcessor.java
+++ b/extensions/websockets/server/deployment/src/main/java/io/quarkus/websockets/deployment/ServerWebSocketProcessor.java
@@ -110,10 +110,11 @@ public class ServerWebSocketProcessor {
         final IndexView index = indexBuildItem.getIndex();
         WebsocketClientProcessor.registerCodersForReflection(reflection, index.getAnnotations(SERVER_ENDPOINT));
 
+        int priority = 1 + FilterBuildItem.AUTHORIZATION;
         return new FilterBuildItem(
                 recorder.createHandler(webSocketDeploymentInfoBuildItem.get().getInfo(),
                         serverWebSocketContainerBuildItem.get().getContainer()),
-                100);
+                priority);
     }
 
     @BuildStep

